packages/block-editor/src/components/media-upload-progress/index.native.js

```diff
@@ -36,10 +39,18 @@ export class MediaUploadProgress extends Component {
 
 	componentDidMount() {
 		this.addMediaUploadListener();
+
+		this.subscription = subscribeConnectionStatus(
+			this.handleConnectionStatusChange
+		);
```

Comment:
Noting that we if we retain the approach of _not_ using the `isConnected` Hook, we'll need to update this `componentDidMount` to fetch the initial state of the network connection, similar to the Hook.

https://github.com/WordPress/gutenberg/blob/aadf8c66ff345b176041ad82e3793191ade5d271/packages/react-native-bridge/index.js#L196-L212